soh/soh/Enhancements/randomizer/randomizer_check_tracker.cpp

```diff
@@ -46,76 +153,662 @@ Color_RGBA8 Color_Skipped_Main_Default              = { 160, 160, 160, 255 };
+        return (gPlayState->sceneNum == scene && gPlayState->actorCtx.flags.chest & (1 << flag)) ||
+               gSaveContext.sceneFlags[scene].chest & (1 << flag);
+    case SpoilerCollectionCheckType::SPOILER_CHK_COLLECTABLE:
+        return (gPlayState->sceneNum == scene && gPlayState->actorCtx.flags.collect & (1 << flag)) ||
+               gSaveContext.sceneFlags[scene].collect & (1 << flag);
+    case SpoilerCollectionCheckType::SPOILER_CHK_MERCHANT:
+    case SpoilerCollectionCheckType::SPOILER_CHK_SHOP_ITEM:
+    case SpoilerCollectionCheckType::SPOILER_CHK_COW:
+    case SpoilerCollectionCheckType::SPOILER_CHK_SCRUB:
+    case SpoilerCollectionCheckType::SPOILER_CHK_RANDOMIZER_INF:
+        return Flags_GetRandomizerInf(OTRGlobals::Instance->gRandomizer->GetRandomizerInfFromCheck(rc));
+    case SpoilerCollectionCheckType::SPOILER_CHK_EVENT_CHK_INF:
+        return gSaveContext.eventChkInf[flag / 16] & (0x01 << flag % 16);
+    case SpoilerCollectionCheckType::SPOILER_CHK_GERUDO_MEMBERSHIP_CARD:
+        return CHECK_FLAG_ALL(gSaveContext.eventChkInf[0x09], 0x0F);
+    case SpoilerCollectionCheckType::SPOILER_CHK_GOLD_SKULLTULA:
+        return GET_GS_FLAGS(scene) & flag;
+    case SpoilerCollectionCheckType::SPOILER_CHK_INF_TABLE:
+        return gSaveContext.infTable[scene] & INDEX_TO_16BIT_LITTLE_ENDIAN_BITMASK(flag);
+    case SpoilerCollectionCheckType::SPOILER_CHK_ITEM_GET_INF:
+        return gSaveContext.itemGetInf[flag / 16] & INDEX_TO_16BIT_LITTLE_ENDIAN_BITMASK(flag);
+    case SpoilerCollectionCheckType::SPOILER_CHK_MAGIC_BEANS:
+        return BEANS_BOUGHT >= 10;
+    case SpoilerCollectionCheckType::SPOILER_CHK_NONE:
+        return false;
+    case SpoilerCollectionCheckType::SPOILER_CHK_GRAVEDIGGER:
+        // Gravedigger has a fix in place that means one of two save locations. Check both.
+        return (gSaveContext.itemGetInf[1] & 0x1000) || // vanilla flag
+            ((gSaveContext.n64ddFlag || CVarGetInteger("gGravediggingTourFix", 0)) &&
+                gSaveContext.sceneFlags[scene].collect & (1 << flag) || (gPlayState->actorCtx.flags.collect & (1 << flag))); // rando/fix flag
+    default:
+        return false;
+    }
+    return false;
+}
+
+void CheckTrackerDialogClosed() {
+    if (messageCloseCheck) {
+        messageCloseCheck = false;
+    }
+}
+
+void CheckTrackerShopSlotChange(uint8_t cursorSlot, int16_t basePrice) {
+    if (gPlayState->sceneNum != SCENE_HAPPY_MASK_SHOP) { // Happy Mask Shop is not used in rando, so is not tracked
+        auto slot = startingShopItem.find(gPlayState->sceneNum)->second + cursorSlot;
+        auto status = gSaveContext.checkTrackerData[slot].status;
+        if (status == RCSHOW_SEEN) {
+            gSaveContext.checkTrackerData[slot].status = RCSHOW_IDENTIFIED;
+            gSaveContext.checkTrackerData[slot].price = basePrice;
+            SaveManager::Instance->SaveSection(gSaveContext.fileNum, sectionId, true);
+        }
+    }
```

Comment:
formatting nit to avoid nesting

```suggestion
    // Happy Mask Shop is not used in rando, so is not tracked
    if (gPlayState->sceneNum == SCENE_HAPPY_MASK_SHOP) {
        return;
    }

    auto slot = startingShopItem.find(gPlayState->sceneNum)->second + cursorSlot;
    auto status = gSaveContext.checkTrackerData[slot].status;
    if (status == RCSHOW_SEEN) {
        gSaveContext.checkTrackerData[slot].status = RCSHOW_IDENTIFIED;
        gSaveContext.checkTrackerData[slot].price = basePrice;
        SaveManager::Instance->SaveSection(gSaveContext.fileNum, sectionId, true);
    }
```